duckdb_engine/tests/test_basic.py

```diff
@@ -166,6 +166,51 @@ def test_get_tables(inspector: Inspector) -> None:
     assert inspector.get_view_names() == []
 
 
+@mark.skipif(
+    supports_attach is False,
+    reason="ATTACH is not supported for DuckDB version < 0.7.0",
+)
+def test_get_schema_names(inspector: Inspector, engine: Engine) -> None:
+    # Run each command in a separate transaction
+    for cmd in [
+        """CREATE SCHEMA "quack quack" """,
+        """ATTACH ':memory:' AS "daffy duck" """,
+        """CREATE SCHEMA "daffy duck"."quack quack" """,
+        """CREATE TABLE "daffy duck"."quack quack"."t1" (i INTEGER, j INTEGER);""",
+        """CREATE TABLE "daffy duck"."quack quack"."t2" (i INTEGER, j INTEGER);""",
+        """CREATE SCHEMA "daffy duck"."you're "" despicable" """,
+    ]:
+        with engine.connect() as conn:
```

Comment:
Isn't this just creating a new in memory database for every loop iteration?

Reply:
Ah, you are right, It does indeed look like [closing](https://duckdb.org/docs/api/c/connect.html#duckdb_close) the connection deallocates the memory. My bad. The test passes for me locally so will try to repro this first. Thanks